course/src/isometry.cpp

```diff
@@ -180,5 +180,181 @@ namespace math {
   const Vector3 Vector3::kUnitZ = Vector3(0.0, 0.0, 1.0);
   const Vector3 Vector3::kZero = Vector3(0.0, 0.0, 0.0);
 
+  // Matrix --------------------------------------------
+
+  Matrix3::Matrix3(
+    const double a1, const double a2, const double a3,
+    const double b1, const double b2, const double b3,
+    const double c1, const double c2, const double c3) :
+    rows{Vector3{a1, a2, a3}, Vector3{b1, b2, b3}, Vector3{c1, c2, c3}} {}
+
+  Matrix3::Matrix3(
+    const Vector3& row1, const Vector3& row2, const Vector3& row3) :
+    rows{row1, row2, row3} { }
+
+  Matrix3::Matrix3() :
+    rows{
+      Vector3{0.0, 0.0, 0.0}, Vector3{0.0, 0.0, 0.0}, Vector3{0.0, 0.0, 0.0}} {}
+
+  const Matrix3 Matrix3::kIdentity =
+    Matrix3({1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0});
+
+  const Matrix3 Matrix3::kZero =
+    Matrix3({0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0});
+
+  const Matrix3 Matrix3::kOnes =
+    Matrix3({1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0});
```

Comment:
Your `Matrix3` implementation does not have a constructor that receives an `initializer_list`. Question: what's this `{ ... } ` pair really doing then?

Same in the ones above.